### ISSUE TITLE
stretched images (page number)

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#448]
looking at the query data of #447 
the forth data row

    7960847	Orchesterwerke_Romantik_Themen.pdf	67	31111	1044861	15594
refers to a .pdf
This occurs in 18 different pages
6 times in en.wikipedia.org/wiki/Symphony_No._1_(Brahms)
Looking at that page
![brahms1](https://user-images.githubusercontent.com/861631/57008500-8cc7cd80-6be8-11e9-8834-7e6233c8d252.jpg)
It can be seen that some of the images are distorted
The wikitext is

    ==Structure==
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=84]]
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=85]]
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=86]]
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=87]]
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=88]]
    [[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=89]]

I think this issue is related to #440 in that the 'page' is not being used correctly (or displayed correctly)
Each thumb should be unique in the file-core database
(strictly the thumb is but the reference should take 'page' into consideration somehow)
The fil_name  in fsdb_fil should be unique
Perhaps
Orchesterwerke Romantik Themen.pdf/84
or
Orchesterwerke Romantik Themen.pdf/page84
Only different sizes of the same pdf and page number should be combined

If the suggestion in #447 is to be implemented, this issue needs to be resolved first

**Comment by desb42** [comment]
As wikisources use the 'page' mechanism a lot more than the other wikis
pdfs and djvus
I think this is quite a significant issue for wikisources

**Comment by gnosygnu** [comment]
> It can be seen that some of the images are distorted

Yeah, the problem is with the height. XOWA is using 134, when it should be 104

```
[[File:Orchesterwerke Romantik Themen.pdf|right|300px|page=84]]

<a href="/wiki/File:Orchesterwerke_Romantik_Themen.pdf" class="image" xowa_title="Orchesterwerke_Romantik_Themen.pdf"><img id="xoimg_5" alt="" src="file:///C:/xowa/file/commons.wikimedia.org/thumb/7/e/2/e/Orchesterwerke_Romantik_Themen.pdf/300px-84.jpg" width="300" height="134" /></a></div>
```

Specifically, for pdfs, XOWA uses an aspect ratio from the first page of the pdf: https://commons.wikimedia.org/w/api.php?action=query&format=xml&prop=imageinfo&iiprop=size|url&redirects&titles=File:Orchesterwerke_Romantik_Themen.pdf

In this case, it's 2666 by 1189 which is the same as 300 x 134. However, the file itself is 300 x 100. Hence the distortion.

XOWA relies on the wmf meta data b/c it's more easily accessible than the actual image data (I'd have to use ImageMagick, which is slower). I'll have to look at the api / image dumps to see if they give metadata by page, but I think that's unlikely. So I'll probably have to go the imageMagick way.

> Each thumb should be unique in the file-core database
> (strictly the thumb is but the reference should take 'page' into consideration somehow)
> The fil_name in fsdb_fil should be unique
> Perhaps
> Orchesterwerke Romantik Themen.pdf/84
> or
> Orchesterwerke Romantik Themen.pdf/page84

Actually, this does exist to some extent. There is a column in the fsdb_thm table called thm_page which uniquefies by page_number. The same for thm_time (for movie thumbnails).

This table is for thumbs. I don't think it makes sense to do it for files (originals) because Wikimedia doesn't expose the metadata for pages (only metadata for the pdf). I forget what MediaWiki uses to generate thumbnails for pdf pages, but it's an on-the-fly process with no meta-data impact.

>As wikisources use the 'page' mechanism a lot more than the other wikis pdfs and djvus
> I think this is quite a significant issue for wikisources

Yeah, I'll have to figure out a solution to the metadata issue above.
